src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/TypesTests.cs

```diff
@@ -14,23 +16,35 @@ public class TypesTests
     {
         private static IJvmBridge Jvm { get; } = SparkEnvironment.JvmBridge;
 
+        public class SimpleTypesGenerator : IEnumerable<object[]>
+        {
+            private readonly List<object[]> _data = new List<object[]>
+            {
+                new object[] { new NullType() },
+                new object[] { new StringType() },
+                new object[] { new BinaryType() },
+                new object[] { new BooleanType() },
+                new object[] { new DateType() },
+                new object[] { new TimestampType() },
+                new object[] { new DoubleType() },
+                new object[] { new FloatType() },
+                new object[] { new ByteType() },
+                new object[] { new IntegerType() },
+                new object[] { new LongType() },
+                new object[] { new ShortType() },
+                new object[] { new DecimalType() }
+            };
+
+            public IEnumerator<object[]> GetEnumerator() => _data.GetEnumerator();
+
+            IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+        }
+
         [Theory]
-        [InlineData("null")]
-        [InlineData("string")]
-        [InlineData("binary")]
-        [InlineData("boolean")]
-        [InlineData("date")]
-        [InlineData("timestamp")]
-        [InlineData("double")]
-        [InlineData("float")]
-        [InlineData("byte")]
-        [InlineData("integer")]
-        [InlineData("long")]
-        [InlineData("short")]
-        public void TestSimpleTypes(string typeName)
+        [ClassData(typeof(SimpleTypesGenerator))]
+        public void TestSimpleTypes(DataType simpleTypes)
         {
-            JvmObjectReference atomicType = DataType.FromJson(Jvm, $@"""{typeName}""");
-            Assert.IsType<JvmObjectReference>(atomicType);
+            Assert.IsType<JvmObjectReference>(DataType.FromJson(Jvm, simpleTypes.Json));
```

Comment:
How about just create
```C#
private void Validate(DataType dataType)
{
    Assert.IsType<JvmObjectReference>(DataType.FromJson(Jvm, dataType.Json));
}
```

and just
```C#
Validate(new NullType());
Validate(new BinaryType());
```

Reply:
You can reuse this in `TestArrayType`, etc.

Reply:
Do we need separate `[Fact]`s for ArrayType and MapType?  Can the type tests just be contained in one single test method ?

Reply:
yea we discussed this earlier and these will be merged into one method.

Reply:
Thanks for the suggestion! I have changed accordingly.

Reply:
sorry, now that we have one function, we can put at the end of the function:
```C#
            // The following validates for all ComplexTypes.
            Validate(new ArrayType(new IntegerType()));
            Validate(new MapType(new IntegerType(), new StringType()));
            Validate(new StructType(new[]
            {
                new StructField("age", new IntegerType()),
                new StructField("name", new StringType())
            }));

            // StructField is not tested because it cannot be converted from JSON by itself.
        }
```

Reply:
Sure, moved inside the function. Thanks.